src/main/kotlin/app/revanced/patches/music/ad/video/fingerprints/ShowMusicVideoAdsConstructorFingerprint.kt

```diff
@@ -0,0 +1,39 @@
+package app.revanced.patches.music.ad.video.fingerprints
+
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.extensions.or
+import app.revanced.patcher.fingerprint.method.impl.MethodFingerprint
+import app.revanced.patcher.fingerprint.method.annotation.FuzzyPatternScanMethod
+import app.revanced.patcher.fingerprint.method.annotation.MatchingMethod
+import app.revanced.patches.music.ad.video.annotations.MusicVideoAdsCompatibility
+import org.jf.dexlib2.AccessFlags
+import org.jf.dexlib2.Opcode
+
+@Name("show-video-ads-constructor-fingerprint")
+@MatchingMethod(
+    "Laghd",
+    "<init>",
+)
+@FuzzyPatternScanMethod(2) // FIXME: Test this threshold and find the best value.
+@MusicVideoAdsCompatibility
+@Version("0.0.1")
+object ShowMusicVideoAdsConstructorFingerprint : MethodFingerprint(
+    "V", AccessFlags.PUBLIC or AccessFlags.CONSTRUCTOR, listOf("L", "L", "L"), listOf(
+        Opcode.INVOKE_DIRECT,
```

Comment:
too many opcodes, i recommend you using less so it has a higher chance of surviving across new versions.

Reply:
I suppose for now this suffices. It is not too big.